File: app/src/main/java/com/fonfon/noloss/ui/newdevice/NewDeviceView.java
package com.fonfon.noloss.ui.newdevice;

import android.location.Location;
import android.util.Pair;

import com.fonfon.noloss.lib.ActivityEvent;
import com.fonfon.noloss.viewstate.NewDevicesViewState;
import com.hannesdorfmann.mosby3.mvp.MvpView;

import io.reactivex.Observable;

public interface NewDeviceView extends MvpView {

  Observable<ActivityEvent> onLifecycleIntent();

  Observable<Object> onRefreshIntent();

  Observable<Pair<String, String>> onDeviceIntent();

  Observable<Location> onNewLocation();

  void render(NewDevicesViewState state);
}

File: app/src/main/java/com/fonfon/noloss/ui/newdevice/NewDeviceView.kt
package com.fonfon.noloss.ui.newdevice

import android.location.Location
import android.util.Pair
import com.fonfon.noloss.NewDevicesViewState

import com.fonfon.noloss.lib.ActivityEvent
import com.hannesdorfmann.mosby3.mvp.MvpView

import io.reactivex.Observable

interface NewDeviceView : MvpView {

  fun onLifecycleIntent(): Observable<ActivityEvent>

  fun onRefreshIntent(): Observable<Any>

  fun onDeviceIntent(): Observable<Pair<String, String>>

  fun onNewLocation(): Observable<Location>

  fun render(state: NewDevicesViewState)
}
